tools/dotnet-linker/Steps/ManagedRegistrarStep.cs

```diff
@@ -990,6 +948,20 @@ bool IsOpenType (TypeReference tr)
 			return IsOpenType (tr.Resolve ());
 		}
 
+		TypeReference ReplaceOpenTypeIfNeeded (TypeReference type, TypeReference replacementType)
+		{
+			if (IsOpenType (type)) {
+				if (type is ArrayType arrayType) {
+					var elementType = ReplaceOpenTypeIfNeeded (arrayType.ElementType, replacementType);
+					return new ArrayType (elementType, arrayType.Rank);
+				}
+
+				return replacementType;
```

Comment:
What about out/ref parameters?

Reply:
I added the implementation in the last commit a moment ago.